Caltech-IPAC/firefly
# DM-16476: Create a TableTypeWatcher to support monitoring tables for full life time

Create a TableTypeWatcher to support monitoring tables for  full life time

 - Add TypeTypeWatcher system to MasterSaga
 - Modified three watchers for new system: CatalogWatcher, ImageMetaDataWatcher, CoverageWatcher
 - Added more testing
 - fixed ibeQuery attributes, finished ImageMetaDataWatcher
 - CoverageWatcher fix to filter out duplicates for query columns (from tatiana)
 - includes fixed IRSA-2568
 - a little cleanup of DrawLayerCntlr, removed unused action
 - Added viewer test and new viewer test category

_To test:_

https://irsawebdev9.ipac.caltech.edu/DM-16476-master-watcher/firefly/
https://irsawebdev9.ipac.caltech.edu/DM-16476-master-watcher/firefly/firefly-dev.html
https://irsawebdev9.ipac.caltech.edu/DM-16476-master-watcher/firefly/test/tests-main.html

 - Since this is a refactor ticket. Everything should work life before.
 - _Run 1_: bring up some images at a target then load a catalog.  Should see the overlay.
 - _Run 2_: start with catalog and do search should see the coverage like before
 - _Run 3_: 
    - load firefly-dev.html
    - Test Searches => Wise Search => Any pixel overlaps search region => 5.1 deg => AllWise
    - Should see wise coverage and wise metadata tags
 - _Run 4_: load firefly/test/tests-main.html, confirm test 5 is correct.
